src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -245,33 +245,58 @@ module Buffer = struct
          List.rev !lines)
 end
 
-let special_dir_for_fs_sync =
-  lazy
-    (Path.Build.relative Path.Build.root ".sync" |> Path.build |> Path.to_string)
-
-let emit_sync t =
-  let id = Sync_id.gen () in
-  let fn = id |> Sync_id.to_int |> string_of_int in
-  let path = Filename.concat (Lazy.force special_dir_for_fs_sync) fn in
-  Unix.close (Unix.openfile path [ O_WRONLY; O_CREAT; O_TRUNC ] 0o666);
-  Table.set t.sync_table fn id;
-  id
-
-let is_special_file_for_fs_sync ~path_as_reported_by_file_watcher =
-  (* We use string matching here and that's fine because we match on the path
-     reported by the file watcher backend which will always report the same
-     string that was used to setup the watch. *)
-  Filename.dirname path_as_reported_by_file_watcher
-  = Lazy.force special_dir_for_fs_sync
-
-let consume_sync_event table path =
-  let basename = Filename.basename path in
-  match Table.find table basename with
-  | None -> None
-  | Some id ->
-    Fpath.unlink_no_err path;
-    Table.remove table basename;
-    Some id
+module Fs_sync : sig
```

Comment:
I'm confused by changes in this module. They seem to have nothing to do with `Svar`s. Shall this be moved to a separate PR?

Reply:
I tried doing that but with no success (https://github.com/ocaml/dune/pull/5363). We need this commit for this PR to be properly tested, but rpc itself is a little buggy and needs the fixes in this PR. There's an individual commit for it but I can include it in the PR description as well.

Reply:
Got it.